src/controllers/file.controller.js

```diff
@@ -0,0 +1,34 @@
+const httpStatus = require('http-status');
+const catchAsync = require('../utils/catchAsync');
+const multer = require('multer');
+const middlewareUpload = require('../middlewares/upload');
+const config = require('../config/config');
+
+const upload = catchAsync(async (req, res, next) => {
+  middlewareUpload.single('fileName')(req, res, (err) => {
+    if (!err) {
+      if (req.file) {
+        res.status(httpStatus.OK).send(`File is saved!`);
+      } else { // there was no file being uploaded
+        res.status(httpStatus.BAD_REQUEST).send('Please select a file to upload!');
+      }
+    } else {
+      if (err instanceof multer.MulterError) {
+        // A Multer error occurred when uploading.
+        let message = err.message;
+        // other error message from Multer can be used.
+        if (err.code === 'LIMIT_FILE_SIZE') {
+          message = `Maximum file size is ${config.file.maxUploadSize/(1024 * 1024)}MB`;
+        }
+        res.status(httpStatus.BAD_REQUEST).send(message);
+      } else {
+        // An unknown error occurred when uploading.
+        res.status(err.statusCode).send(err.message);
```

Comment:
Saw that, if I don't create a public folder in root, then it just crashes with a statusCode error, which is because of the err.statusCode being undefined/null. So In this else block you should add this part : 
```
// An unknown error occurred when uploading.
          logger.error(err.message);
          if (err.statusCode != null) {
            res.status(err.statusCode).send(err.message);
          } else {
            res.status(httpStatus.INTERNAL_SERVER_ERROR).send(err.message);
          }
```
I've seen the official multer docs, and the example creates a public folder on initialization, maybe you can add that to solve this issue? But for now I guess this can be added to handle the error. As I just spent almost an hour finding where it was coming from lol. Otherwise, it's working like a charm. Thanks <3

Reply:
Thanks a lot for your help and suggestion @SharifRafid :)
I will rectify it and push again :D

Reply:
codes have been rectified and pushed